### DISCUSSION TITLE
user-guide/misc/link-citation-to-bibliography

### DISCUSSION THREAD

**Discussion by giscus** [zotero-chinese/wiki#320] [文档评论区]
# user-guide/misc/link-citation-to-bibliography

Zotero 非官方中文维护小组

https://zotero-chinese.com/user-guide/misc/link-citation-to-bibliography

**Comment by Grant-H-Cq** [comment]
建议在一些情况下，可以将宏代码第170行和174行删除或注释掉，避免在MS word中引用位置附近有公式和符号时，运行脚本后出现公式与周围文本不对齐的bug。

**Reply by Grant-H-Cq** [reply]
行170：style = Selection.style
行174：Selection.style = style

**Comment by EdgerHao** [comment]
运行宏时，弹出：下标越界
如何解决呢

**Reply by seven979** [reply]
你好，请问你解决了吗？我也遇到了这个问题

**Reply by SH-alaways** [reply]
我也出现了这个错误，有人解决了吗

**Reply by SH-alaways** [reply]
已解决 看下面评论

**Comment by SH-alaways** [comment]
出现“运行时错误‘9’，下标越界”这个错误的原因通常是由于访问了数组中不存在的元素，即尝试访问数组的索引超出了其有效范围。人话：一次引用的文献大于宏代码中规定的32个
把Dim array_RefTitle(32) As String中的32改成你的最大引用文献数或者大于引用文献数即可
即：我一次最大引用数为34，我只需要把32改为34即可

**Reply by YouCanRich2** [reply]
请问这个是报错之后直接在弹出来的宏页面里面找的这一段吗？我怎么没有找到这个代码呢

**Comment by Goodworkk** [comment]
目前硕士毕业论文用的前面比较简单的宏定义方法，也就是单向超链接；Python方法还没用，硕论用的顺序引用是不是没法使用Python实现了啊

**Comment by aoweidd** [comment]
访问超链接后会出现下划线，没办法取消，怎么解决

**Reply by YouCanRich2** [reply]
可以看一下这位的教程https://b23.tv/xB6KyKT，从05:07开始看

**Comment by murenzzj** [comment]
macos能否使用

**Comment by klxiaoniu** [comment]
https://gitee.com/wangrui5015/Zotero-WPSJS

Zotero-WPSJS有个Citation Hyperlinks的功能，可以创建引注到参考文献的超链接

**Comment by YouCanRich2** [comment]
大家一定一定一定一定一定一定要在终稿已经确定完全没有的时候，最后的最后再开始弄这个超链接，我拿我的初稿建立了超链接之后给导师看，导师让我返修，删掉和调换了一些文献，再想建立链接的时候就出现了“手动修改引文链接或删除引号可能导致「下标超出范围（Subscript out of range）」错误”的情况，101篇文献啊！！！我按照作者的方法一个个点，人都崩溃了：(

**Reply by YouCanRich2** [reply]
作者你知道我看到你说“此问题几乎无法通过 VBA 脚本自动解决，目前也没有办法强制 Zotero 在所有字段中更新 CSL JSON 数据中引文对象的顺序。”有多崩溃吗

**Comment by lly645483-cloud** [comment]
我的文库列表作者那一栏全是中文的“等”字，引用的时候也是中文的“等”，但文献信息部分我已经把语言改成了en也不行，有人可以解答一下吗？

**Reply by northword** [reply]
你要是用英文的csl样式，然后还要确保word里zotero文档首选项里有个语言选en而不是中文

**Comment by Dyao111111** [comment]
那个宏文件怎么下载？点链接进去是一段代码
